AnnService/src/Core/VectorIndex.cpp

```diff
@@ -294,6 +294,42 @@ VectorIndex::DeleteIndex(ByteArray p_meta) {
 }
 
 
+ErrorCode
+VectorIndex::MergeIndex(const char* p_indexFilePath)
+{
+    std::string folderPath(p_indexFilePath);
+    if (!folderPath.empty() && *(folderPath.rbegin()) != FolderSep) folderPath += FolderSep;
+    Helper::IniReader iniReader;
+    if (ErrorCode::Success != iniReader.LoadIniFile(folderPath + "indexloader.ini")) return ErrorCode::FailedOpenFile;
+
+    std::shared_ptr<VectorIndex> addIndex = CreateInstance(iniReader.GetParameter("Index", "IndexAlgoType", IndexAlgoType::Undefined),
+        iniReader.GetParameter("Index", "ValueType", VectorValueType::Undefined));
+    addIndex->LoadConfig(iniReader);
+    addIndex->LoadIndexData(folderPath);
+
+    std::shared_ptr<MetadataSet> pMetadata;
+    if (iniReader.DoesSectionExist("MetaData"))
+    {
+        pMetadata.reset(new MemMetadataSet(folderPath + iniReader.GetParameter("MetaData", "MetaDataFilePath", std::string()),
+            folderPath + iniReader.GetParameter("MetaData", "MetaDataIndexPath", std::string())));
+    }
+
+#pragma omp parallel for schedule(dynamic,128)
+    for (SizeType i = 0; i < addIndex->GetNumSamples(); i++)
+        if (addIndex->ContainSample(i))
+        {
+            std::shared_ptr<MetadataSet> p_metaSet;
+            if (pMetadata != nullptr) {
+                ByteArray meta = pMetadata->GetMetadata(i);
```

Comment:
can we ensure meta always valid here?

Reply:
no, we allow buildindex without metadata.